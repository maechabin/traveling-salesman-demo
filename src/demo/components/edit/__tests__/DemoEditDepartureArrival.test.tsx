import React from 'react';
import { shallow } from 'enzyme';

import DemoEditDepartureArrival from '../DemoEditDepartureArrival';

describe('<DemoDescription />', () => {
  it('snapshot', () => {
    // setup
    const label = 'AAA';
    const title = 'BBB';
    const index = 0;
    const callbackSpy = jest.fn();
    const props = {
      callback: callbackSpy,
      label,
      title,
      index,
    };

    // exercise
    const wrapper = shallow(<DemoEditDepartureArrival {...props} />);

    // verify
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange', () => {
    // setup
    const label = 'AAA';
    const title = 'BBB';
    const index = 0;
    const callbackSpy = jest.fn();
    const props = {
      callback: callbackSpy,
      label,
      title,
      index,
    };
    const event = 'event';
    const wrapper = shallow(<DemoEditDepartureArrival {...props} />);

    // exercise
    wrapper.find('input').simulate('change', event);

    // verify
    expect(callbackSpy).toHaveBeenCalledWith(event, index);
  });
});
