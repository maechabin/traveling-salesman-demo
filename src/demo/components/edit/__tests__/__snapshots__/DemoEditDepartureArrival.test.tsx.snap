// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DemoDescription /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DemoEditDepartureArrival
    callback={[MockFunction]}
    index={0}
    label="AAA"
    title="BBB"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          style={
            Object {
              "backgroundColor": "#f75c50",
              "boxSizing": "border-box",
              "color": "#fff",
              "display": "block",
              "padding": "4px",
              "textAlign": "center",
              "width": "320px",
            }
          }
        >
          AAA
        </label>,
        <ul
          style={
            Object {
              "listStyleType": "none",
              "margin": "0 0 16px 0",
              "padding": 0,
            }
          }
        >
          <li>
            <input
              defaultValue="BBB"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "AAA",
          "style": Object {
            "backgroundColor": "#f75c50",
            "boxSizing": "border-box",
            "color": "#fff",
            "display": "block",
            "padding": "4px",
            "textAlign": "center",
            "width": "320px",
          },
        },
        "ref": null,
        "rendered": "AAA",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <li>
            <input
              defaultValue="BBB"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />
          </li>,
          "style": Object {
            "listStyleType": "none",
            "margin": "0 0 16px 0",
            "padding": 0,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              defaultValue="BBB"
              onChange={[Function]}
              style={
                Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "BBB",
              "onChange": [Function],
              "style": Object {
                "boxSizing": "border-box",
                "fontSize": "16px",
                "imeMode": "active",
                "padding": "6px 4px",
                "width": "320px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "li",
        },
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            style={
              Object {
                "backgroundColor": "#f75c50",
                "boxSizing": "border-box",
                "color": "#fff",
                "display": "block",
                "padding": "4px",
                "textAlign": "center",
                "width": "320px",
              }
            }
          >
            AAA
          </label>,
          <ul
            style={
              Object {
                "listStyleType": "none",
                "margin": "0 0 16px 0",
                "padding": 0,
              }
            }
          >
            <li>
              <input
                defaultValue="BBB"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "AAA",
            "style": Object {
              "backgroundColor": "#f75c50",
              "boxSizing": "border-box",
              "color": "#fff",
              "display": "block",
              "padding": "4px",
              "textAlign": "center",
              "width": "320px",
            },
          },
          "ref": null,
          "rendered": "AAA",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <input
                defaultValue="BBB"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />
            </li>,
            "style": Object {
              "listStyleType": "none",
              "margin": "0 0 16px 0",
              "padding": 0,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                defaultValue="BBB"
                onChange={[Function]}
                style={
                  Object {
                    "boxSizing": "border-box",
                    "fontSize": "16px",
                    "imeMode": "active",
                    "padding": "6px 4px",
                    "width": "320px",
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "defaultValue": "BBB",
                "onChange": [Function],
                "style": Object {
                  "boxSizing": "border-box",
                  "fontSize": "16px",
                  "imeMode": "active",
                  "padding": "6px 4px",
                  "width": "320px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "li",
          },
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
